acp/identity/identity.go

```diff
@@ -45,14 +46,20 @@ func FromPrivateKey(privateKey *secp256k1.PrivateKey) immutable.Option[Identity]
 // FromPublicKey returns a new identity using the given public key.
 func FromPublicKey(publicKey *secp256k1.PublicKey) immutable.Option[Identity] {
 	return immutable.Some(Identity{
-		Address:   AddressFromPublicKey(publicKey),
+		DID:       DIDFromPublicKey(publicKey),
 		PublicKey: publicKey,
 	})
 }
 
-// AddressFromPublicKey returns the unique address of the given public key.
-func AddressFromPublicKey(publicKey *secp256k1.PublicKey) string {
-	pub := cosmosSecp256k1.PubKey{Key: publicKey.SerializeCompressed()}
-	// conversion from well known types should never cause a panic
-	return types.MustBech32ifyAddressBytes("cosmos", pub.Address().Bytes())
+// DIDFromPublicKey returns the unique address of the given public key.
+func DIDFromPublicKey(publicKey *secp256k1.PublicKey) string {
+	bytes := publicKey.SerializeUncompressed()
+	did, err := key.CreateDIDKey(crypto.SECP256k1, bytes)
+	// because we are generating a did from a a supported and fixed key type,
+	// this operation is infallible
+	if err != nil {
+		panic(err)
+	}
```

Comment:
question: why do we have a panic here? instead of returning an error? also document that in a comment please.

Reply:
Initially I implemented it as an error, however when I started testing the function, I realized the error would never happen under those circumstances. It didn't seem to make sense to return an error if it would never happens. As for the comment, it's comented that the operation is infallible.

Reply:
I have a preference to avoid panics in prod code. If this library `cyware/ssi-sdk/did/key` starts returning errors more frequently we should have proper error handling IMO over a panic.

However can hear from others too, @sourcenetwork/database-team happy to follow consensus. Also incase keeping panic should follow the naming of functions having `Must` key word in their name

Reply:
https://github.com/sourcenetwork/defradb/issues/615

More context.

Discussed a bit in standup and we currently have only 3 panics, 2 of which we want to maintain.

Coming back to our discussion, I have a harder preference for it now.